Packs/SalesforceITAdmin/Integrations/Salesforce_IT_Admin/Salesforce_IT_Admin.py

```diff
@@ -0,0 +1,1164 @@
+            "instanceName": self.instanceName,
+            "success": success,
+            "active": active,
+            "id": iden,
+            "username": username,
+            "email": email,
+            "errorCode": errorCode,
+            "errorMessage": errorMessage,
+            "details": details
+        }
+
+
+''' HELPER FUNCTIONS '''
+
+
+def verify_and_load_scim_data(scim):
+    try:
+        scim = json.loads(scim)
+    except Exception:
+        pass
+    if type(scim) != dict:
+        raise Exception("SCIM data is not a valid JSON")
+    return scim
+
+
+def map_scim(scim):
+    try:
+        scim = json.loads(scim)
+    except Exception:
+        pass
+    if type(scim) != dict:
+        raise Exception('Provided client data is not JSON compatible')
+
+    mapping = {
+        "userName": "userName",
+        "email": "emails(val.primary && val.primary==true).[0].value",
+        "first_name": "name.givenName",
+        "last_name": "name.familyName",
+        "name": "displayName",
+        "nick_name": "nickName",
+        "id": "id",
+        "active": "active",
+        "address": "addresses(val.primary && val.primary==true).formatted",
+        "street": "addresses(val.primary && val.primary==true).streetAddress",
+        "city": "addresses(val.primary && val.primary==true).locality",
+        "country": "addresses(val.primary && val.primary==true).country",
+        "state": "addresses(val.primary && val.primary==true).region",
+        "zip": "addresses(val.primary && val.primary==true).postalCode",
+        "timezone": "timezone",
+        "locale": "locale",
+        "mobile_phone": "phoneNumbers(val.type && val.type=='mobile').value",
+        "phone": "phoneNumbers(val.type && val.type=='work').value",
+        "title": "title",
+        "user_type": "userType",
+        "full_photo_url": "photos(val.type && val.type=='photo').value",
+        "small_photo_url": "photos(val.type && val.type=='thumbnail').value",
+    }
+    parsed_scim = dict()
+    for k, v in mapping.items():
+        try:
+            value = demisto.dt(scim, v)
+            if(type(value) == list):
+                parsed_scim[k] = value[0]
+            else:
+                parsed_scim[k] = value
+        except Exception:
+            parsed_scim[k] = None
+    return parsed_scim
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client, args):
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    Args:
+        client: SalesforceITAdmin client
+        args  : SalesforceITAdmin arguments passed
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    args
+    uri = URI_PREFIX + 'sobjects/User/testid'
+    res = client.http_request(method='GET', url_suffix=uri)
+    if res.status_code == 200 or res.status_code == 404:
+        return 'ok', None, None
+    else:
+        res_json = res.json()[0]
```

Comment:
I'm worried from IndexError exception here (relevant for all the following occurrences of this line in the rest of the commands). Are we 100% sure that we will always get an array of length >= 1 every time the request fail? If not let's handle it properly - better to do it in the functions that make the http call

Reply:
Yes, Json response will always have the array of length >= 1 in case of request fail.